SharedCryptoPkg/PlatformBuildWorker.py

```diff
@@ -0,0 +1,272 @@
+from optparse import OptionParser
+import logging
+import subprocess
+import shutil
+import struct
+import uuid
+from datetime import datetime
+from datetime import date
+import time
+from MuEnvironment.UefiBuild import UefiBuilder
+
+from MuPythonLibrary.UtilityFunctions import RunCmd
+from MuPythonLibrary.UtilityFunctions import RunPythonScript
+
+# --------------------------------------------------------------------------------------------------------
+# Subclass the UEFI builder and add platform specific functionality.
+#
+
+
+class PlatformBuilder(UefiBuilder):
+
+    def __init__(self, WorkSpace, PackagesPath, PInManager, PInHelper, args):
+        super(PlatformBuilder, self).__init__(
+            WorkSpace, PackagesPath, PInManager, PInHelper, args)
+
+    def SetPlatformEnv(self):
+
+        self.env.SetValue(
+            "ACTIVE_PLATFORM", "SharedCryptoPkg/SharedCryptoDriver.dsc", "Platform Hardcoded")
+        self.env.SetValue("PRODUCT_NAME", "SharedCrypto", "Platform Hardcoded")
+        self.env.SetValue(
+            "TARGET_ARCH", "IA32 X64 AARCH64 ARM", "Platform Hardcoded")
+        self.env.SetValue("TARGET", "RELEASE", "Platform Hardcoded")
+
+        self.env.SetValue("LaunchBuildLogProgram", "Notepad",
+                          "default - will fail if already set", True)
+        self.env.SetValue("LaunchLogOnSuccess", "True",
+                          "default - will fail if already set", True)
+        self.env.SetValue("LaunchLogOnError", "True",
+                          "default - will fail if already set", True)
+
+        return 0
+
+    def __SetEsrtGuidVars(self, var_name, guid_str, desc_string):
+        cur_guid = uuid.UUID(guid_str)
+        self.env.SetValue("BLD_*_%s_REGISTRY" %
+                          var_name, guid_str, desc_string)
+        self.env.SetValue("BLD_*_%s_BYTES" % var_name, "{" + (
+            ",".join(("0x%X" % byte) for byte in cur_guid.bytes_le)) + "}", desc_string)
+        return
+
+    def SetPlatformEnvAfterTarget(self):
+        logging.debug("PlatformBuilder SetPlatformEnvAfterTarget")
+        rc = 0
+
+        #
+        # if private/developer local build we must compute version id value.
+        # If server build (public/shared build) the ID will be passed in and the
+        # compute step will be skipped.
+        #
+        try:
+            rc = self.ComputeVersionValue()
+
+        # value error thrown when build id is outside valid range.  Should never happen unless logic is flawed
+        except ValueError as arg:
+            logging.critical(
+                "ComputeVersionValue created invalid value.  " + str(arg))
+            return -5
+
+        if(rc != 0):
+            logging.critical("ComputeVersionValue failed")
+            return rc
+
+        #
+        # Validate that the value is correct for this product.
+        #
+        rc = self.ValidateVersionValue()
+        if(rc != 0):
+            logging.critical("ValidateVersionValue failed")
```

Comment:
this "version" stuff is not valid for this build

Reply:
Fixed.